src/components/armed_grenade.rs

```diff
@@ -0,0 +1,129 @@
+use macroquad::{
+    color,
+    experimental::{
+        animation::{AnimatedSprite, Animation},
+        collections::storage,
+        scene::{Node, RefMut},
+    },
+    math::{vec2, Vec2},
+    prelude::*,
+};
+
+use crate::{components::PhysicsBody, Resources};
+
+pub struct ArmedGrenade {
+    grenade_sprite: AnimatedSprite,
+    pub body: PhysicsBody,
+    lived: f32,
+}
+
+impl ArmedGrenade {
+    pub const COUNTDOWN_DURATION: f32 = 0.5;
+    pub const EXPLOSION_RADIUS: f32 = 100.0;
+
+    pub fn new(pos: Vec2, facing: bool) -> Self {
+        // TODO: In case we want to animate thrown grenades rotating etc.
+        let grenade_sprite = AnimatedSprite::new(
+            21,
+            28,
+            &[Animation {
+                name: "idle".to_string(),
+                row: 1,
+                frames: 4,
+                fps: 8,
+            }],
+            false,
+        );
+
+        let speed = if facing {
+            vec2(600., -200.)
+        } else {
+            vec2(-600., -200.)
+        };
+
+        let mut resources = storage::get_mut::<Resources>();
+
+        let grenade_mount_pos = if facing {
+            vec2(30., 10.)
+        } else {
+            vec2(-50., 10.)
+        };
+
+        let size = vec2(15.0, 15.0);
+
+        let body = PhysicsBody {
+            pos,
+            facing,
+            angle: 0.0,
+            size,
+            speed,
+            collider: resources.collision_world.add_actor(
+                pos + grenade_mount_pos,
+                size.x as _,
+                size.y as _,
+            ),
+            on_ground: false,
+            last_frame_on_ground: false,
+            have_gravity: true,
+            bouncyness: 0.5,
+        };
+
+        ArmedGrenade {
+            grenade_sprite,
+            body,
+            lived: 0.0,
+        }
+    }
+
+    pub fn spawn(pos: Vec2, facing: bool) {
+        let grenade = ArmedGrenade::new(pos, facing);
+        scene::add_node(grenade);
+    }
+}
+
+impl Node for ArmedGrenade {
```

Comment:
The idea of `components` - its a piece of logic that may help building nodes (described here: https://github.com/fishfight/FishFight/blob/main/src/components.rs#L1)

As for me, your `ArmedGrenade` struct belong more to an Item?

Reply:
good point, at the time I was associating it with bullets which was a component to make firearms. ArmedGrenade is prety much the projectile of grenades but also there is not a common explosive component so it makes it a not a proper component, but similar to an item at the moment.

Reply:
Bullets lives in their items, like muscet's bullet lives in muscet.rs, so maybe put ArmedGrenade near Grenade item as well?

Reply:
I don't think I follow that.

Reply:
ArmedGrenade is not an item by itself, it is an implementation detail of a Grenade, an item. 

Therefore it belongs to a /grenade/armed_grenade.rs
or just in grenade.rs

Just as muscet's bullet lives in muscet.rs

Reply:
Oh, I get what you mean now, yeah that makes sense and should work better that way